ArmPkg/Drivers/ArmGicDxe/GicV3/ArmGicV3Dxe.c

```diff
@@ -739,6 +739,22 @@ GicV3ExitBootServicesEvent (
   ArmGicDisableDistributor (mGicDistributorBase);
 }
 
+#ifdef MDE_CPU_ARM
+
+/**
+  Dummy GICv5 init funtion to avoid symbol resolution failure on ARM,
+  which cannot support GICv5.
+ **/
+EFI_STATUS
+GicV5DxeInitialize (
+  VOID
+  )
+{
+  return EFI_UNSUPPORTED;
+}
+
+#endif
+

```

Comment:
I intentionally kept this confined to the combined v2/v3 driver so that the GicV3+ driver is legacy free.

In spite of the #ifdef, this just creates technical debt by adding ARM specific code to a source file that was generic before.

Reply:
> Whereas the other puts GicV5 code, and GicV5 source files, into an architecture that cannot support it.
> 

Yes, but under an `Arm/` sub-directory that never gets accessed by the build unless building for ARM.

> One way clutters and confuses the source tree. The other way clutters a corner of a single source file, at the only point it's needed.
> 
> The only debt that is carried is that which the driver has intentionally opted into, only more obviously. I'm happy for that chunk to be dropped outright at any point, and there is no plausible situation in which it could be expanded.

I strongly feel that it should not exist to begin with. I'd rather start removing ARM code entirely than adding new `#ifdef MDE_CPU_ARM` checks to AArch64 code.

Reply:
... but the file does not need to be called `Arm/GicV5/GicV5c.` though - we can rename it to something more appropriate.